nck/readers/ttd_reader.py

```diff
@@ -72,26 +67,17 @@ def the_trade_desk(**kwargs):
 
 class TheTradeDeskReader(Reader):
     def __init__(
-        self,
-        login,
-        password,
-        advertiser_id,
-        report_template_name,
-        report_schedule_name,
-        start_date,
-        end_date,
-        normalize_stream,
+        self, login, password, advertiser_id, report_template_name, report_schedule_name, start_date, end_date, date_range,
     ):
         self.login = login
         self.password = password
         self._build_headers()
         self.advertiser_ids = list(advertiser_id)
         self.report_template_name = report_template_name
         self.report_schedule_name = report_schedule_name
-        self.start_date = start_date
+        self.start_date, self.end_date = build_date_range(start_date, end_date, date_range)
         # Report end date is exclusive: to become inclusive, it should be incremented by 1 day
-        self.end_date = end_date + timedelta(days=1)
-        self.normalize_stream = normalize_stream
+        self.end_date = self.end_date + timedelta(days=1)
 
         self._validate_dates()
```

Comment:
I think that we don't need the `self._validate_dates()` check anymore (as we are already checking dates conformity in your  newly implemented `build_date_range()` function ) 😊